Bottom: f4df5cd7096c28bb6e2005b9b32605ef5eef99cc
Top:    88254c34bb3c28ee3d003c0ba759f2a933f97a81
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-15 05:01:28 +0100

Try optimising polygon operations by keeping an rtree with their contours

InsertHoles() already creates an rtree of contours to do its internal
processing, and it is possible that intersect() could be sped up by
its existance as well.




---

diff --git a/src/polyarea.h b/src/polyarea.h
index 42acc1a..819b074 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -128,6 +128,7 @@ struct POLYAREA
 {
     POLYAREA *f, *b;
     PLINE *contours;
+    void *contour_tree;
 };
 
 BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
